draft-sheffer-wimse-s2s-protocol.md

```diff
@@ -138,6 +138,27 @@ This document uses "service" and "workload" interchangeably. Otherwise, all term
 
 # Using Mutual TLS for Service To Service Authentication {#mutual-tls}
 
+The WIMSE workload identity may be carried within an X.509 certificate. The WIMSE workload identity MUST be encoded in a SubjctAltName extension of type URI.  There MUST be only one SubjectAltName extension of type URI in a WIMSE certificate.  The WIMSE certificate may contain SubjectAltName extensions of other types such as DNSName.
+
+WIMSE identities may be used to validate server and client connections.  When validating a WIMSE identity the relying party MUST validate that the CA issuer for the WIMSE identity is authorized to issue certificates for the trust domain of the WIMSE identity in the certificate. Other PKIX path validation rules apply.
+
+Servers wishing to use the WIMSE identity for authorizing the client MUST require client certificate authentication in the TLS handshake. Other methods of post handshake authentication are not specified by this document.
+
+WIMSE clients and servers MUST validate that the trust domain portion of the WIMSE certificate matches the expected trust domain for the other side of the connection.
+
+## Host Name Validation
+
+[TODO: the following paragraph needs better alignment with RFC 9525. The following is a very drafty straw man]
+
+ It is RECOMMENDED that the client match the WIMSE identity in the certificate against the WIMSE identity of the workload of the intended server. In this case the host portion of the URI is NOT treated as a host name as specified in section 6.4 of {{!RFC9525}} but rather as a trust domain. The server identity is encoded in the path portion of the WIMSE identity in a deployment specific way.
```

Comment:
This also applies to the server performing client authentication, doesn't it? Or is this section more about the fact that server authentication is performed based on URI SAN and not DNS SAN?

Reply:
I just re-read the last paragraph and it seems to me like a breaking change to many existing TLS stacks (in other words: how is it working out for SPIFFE?). We probably need a deeper discussion.

Reply:
Yes, I struggled with this part as well.   Maybe we require both a DNS SAN and a WIMSE URI SAN?